### PR TITLE
LPD-20233 Add integration tests

### PR DESCRIPTION
**Pull request by Mikellorza**
Included subtask:

1. https://liferay.atlassian.net/browse/LPD-20296 Refactor code to do not show the icon in the header if we are not in edition view
2. https://liferay.atlassian.net/browse/LPD-20233 Add integration tests

### CODE REVIEW

File: modules/apps/journal/journal-test/src/testIntegration/java/com/liferay/journal/web/internal/product/navigation/control/menu/test/EditArticleHeaderProductNavigationControlMenuEntryTest.java [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+		MockHttpServletRequest mockHttpServletRequest =
+			new MockHttpServletRequest();
+
+		MockLiferayPortletRenderRequest mockLiferayPortletRenderRequest =
+			_getMockLiferayPortletRenderRequest(mockHttpServletRequest);
+
+		mockLiferayPortletRenderRequest.setAttribute(
+			PortletServlet.PORTLET_SERVLET_REQUEST, mockHttpServletRequest);
+
+		MVCPortlet mvcPortlet = (MVCPortlet)_portlet;
```
Comment (adolfopa) [inline modules/apps/journal/journal-test/src/testIntegration/java/com/liferay/journal/web/internal/product/navigation/control/menu/test/EditArticleHeaderProductNavigationControlMenuEntryTest.java:148]: Q: Why are you casting this to `MVCPortlet`? It shouldn't be necessary.

File: modules/apps/journal/journal-test/src/testIntegration/java/com/liferay/journal/web/internal/product/navigation/control/menu/test/EditArticleHeaderProductNavigationControlMenuEntryTest.java [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+
+		mvcPortlet.render(
+			mockLiferayPortletRenderRequest,
+			new MockLiferayPortletRenderResponse());
```
Comment (adolfopa) [inline modules/apps/journal/journal-test/src/testIntegration/java/com/liferay/journal/web/internal/product/navigation/control/menu/test/EditArticleHeaderProductNavigationControlMenuEntryTest.java:152]: Why are you invoking the portlet? If you're doing it only to fill in request attributes, why not inject explicitly the display contexts in the test? For what I can see, those are the only things the control panel entry depends on. Making the whole portlet execute a render feels a bit overkill.

**Review by adolfopa — CHANGES_REQUESTED** [review]

File: modules/apps/journal/journal-test/src/testIntegration/java/com/liferay/journal/web/internal/product/navigation/control/menu/test/EditArticleHeaderProductNavigationControlMenuEntryTest.java (hunk 1 above)
Comment (Mikellorza) [reply modules/apps/journal/journal-test/src/testIntegration/java/com/liferay/journal/web/internal/product/navigation/control/menu/test/EditArticleHeaderProductNavigationControlMenuEntryTest.java:148]: I thought it was necessary but I am seeing that no, so , I have deleted this casting.

thanks

File: modules/apps/journal/journal-test/src/testIntegration/java/com/liferay/journal/web/internal/product/navigation/control/menu/test/EditArticleHeaderProductNavigationControlMenuEntryTest.java (hunk 2 above)
Comment (Mikellorza) [reply modules/apps/journal/journal-test/src/testIntegration/java/com/liferay/journal/web/internal/product/navigation/control/menu/test/EditArticleHeaderProductNavigationControlMenuEntryTest.java:152]: Yes, I am doing that to fill the JournalDisplayContext that it is used in the `EditArticleHeaderProductNavigationControlMenuEntry`, What is your idea? . I wouldn't like to create the display context instance with reflection , I prefer to render the portlet and delegate to create things that the test need to execute it. I think that is more maintainable.

Comment (adolfopa) [reply modules/apps/journal/journal-test/src/testIntegration/java/com/liferay/journal/web/internal/product/navigation/control/menu/test/EditArticleHeaderProductNavigationControlMenuEntryTest.java:152]: Oh, I see. The display context is in another module. Makes sense then.

**Review by adolfopa — APPROVED** [review]

**Comment by Mikellorza** [comment]
ci:forward

**Comment by ambrinchaudhary** [comment]
ci:reevaluate:1387401_10458

**Comment by Mikellorza** [comment]
ci:forward:force

**Pull request closed without merging**